lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use `where.missing(...)` to find missing relationship records.
+      #
+      # This cop is enabled in Rails 6.1 or higher.
+      #
+      # @example
+      #   # bad
+      #   Post.left_joins(:author).where(authors: { id: nil })
+      #
+      #   # good
+      #   Post.where.missing(:author)
+      #
+      class WhereMissing < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `where.missing(:%<association>s)` instead of ' \
+              '`%<left_joins_method>s(:%<association>s).where(%<association>ss: { id: nil })`.'
+        RESTRICT_ON_SEND = %i[left_joins left_outer_joins].freeze
+
+        minimum_target_rails_version 6.1
+
+        # @!method where_node_and_argument(node)
+        def_node_search :where_node_and_argument, <<~PATTERN
+          $(send ... :where (hash <(pair $(sym _) (hash (pair (sym :id) (nil))))...> ))
+        PATTERN
+
+        # @!method missing_relationship(node)
+        def_node_search :missing_relationship, <<~PATTERN
+          (pair (sym _) (hash (pair (sym :id) (nil))))
+        PATTERN
+
+        def on_send(node)
+          return unless node.first_argument.sym_type?
+
+          where_node_and_argument(root_receiver(node)) do |where_node, where_argument|
+            next unless same_relationship?(where_argument, node.first_argument)
+
+            range = range_between(node.loc.selector.begin_pos, node.loc.expression.end_pos)
+            register_offense(node, where_node, range)
+            break
+          end
+        end
+
+        private
+
+        def root_receiver(node)
+          node&.parent&.send_type? ? root_receiver(node.parent) : node
+        end
+
+        def same_relationship?(where, left_joins)
+          "#{left_joins.value}s" == where.value.to_s
```

Comment:
It's possible for the `left_joins.value` to be exactly the same as the `where.value` (e.g. no need for the `s` - in has_many relationships). Of course there are other cases where the table will be named different than the relation itself, but I guess we can do little about those cases

Reply:
I see. Indeed it is, I did consider cases like has_many relationships, where the `s` is not needed.
I have also added a test.  https://github.com/rubocop/rubocop-rails/blob/818039ec80ee9ade8c0afe6e9591bd052ce33fe5/spec/rubocop/cop/rails/where_missing_spec.rb#L16-L25